app/DynamicsAdapter/DynamicsAdapter.Web/Health/StatusReasonHealthCheck.cs

```diff
@@ -54,19 +54,47 @@ private async Task<bool> CheckStatusReason(CancellationToken cancellationToken)
             return true;
         }
 
-
-        private async Task<bool> CheckOptionSet(CancellationToken cancellationToken)
+        private IEnumerable<Enumeration> GetListOfOptions (string entityName)
         {
+            IEnumerable<Enumeration> optionsList;
+            switch (entityName)
+            {
+                case "ssg_canadianprovincecodesimpletype":
+                    optionsList= Enumeration.GetAll<CanadianProvinceType>();
+                    break;
+                case "ssg_identificationtypes":
+                    optionsList = Enumeration.GetAll<IdentificationType>();
+                    break;
+                case "ssg_informationsourcecodes":
+                    optionsList = Enumeration.GetAll<InformationSourceType>();
+                    break;
+                case "ssg_addresscategorycodes":
+                    optionsList = Enumeration.GetAll<LocationType>();
+                    break;
+                case "ssg_telephonenumbercategorycodes":
+                    optionsList = Enumeration.GetAll<TelephoneNumberType>();
+                    break;
+                default:
+                    optionsList= Enumeration.GetAll<IdentificationType>();
+                    break;
+            }
+            return optionsList;
+        }
 
-            var idTypes = await _optionSetService.GetAllOptions("ssg_identificationtypes", cancellationToken);
+        private async Task<bool> CheckOptionSet(CancellationToken cancellationToken, List<string> optionTypes)
+        {
 
-            foreach (var identificationType in Enumeration.GetAll<IdentificationType>())
+            foreach (var optionType in optionTypes)
             {
-                if (!idTypes.Any(x => x.Value == identificationType.Value && string.Equals(x.Name, identificationType.Name, StringComparison.OrdinalIgnoreCase)))
-                    return false;
+                var types = await _optionSetService.GetAllOptions(optionType, cancellationToken);
+                foreach (var entityType in GetListOfOptions(optionType))
+                {
+                    if (!types.Any(x => x.Value == entityType.Value && string.Equals(x.Name, entityType.Name, StringComparison.OrdinalIgnoreCase)))
+                        return false;
```

Comment:
can we add some log here, then from log, we can know which optionSet is mismatched.

Reply:
good catch. log added